# Include Slackware specific compile flags, provide symlinks for
# backwards compatibility and build also libqhull.so
# Copyright (C) 2016-2022 Georgi D. Sotirov <gdsotirov@gmail.com>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
# Visit SlackPack at https://sotirov-bg.net/slackpack/
#
diff -urNad qhull-2020.2-orig/Makefile qhull-2020.2/Makefile
--- qhull-2020.2-orig/Makefile	2020-09-04 04:45:49.000000000 +0300
+++ qhull-2020.2/Makefile	2022-07-02 19:49:21.250956833 +0300
@@ -134,11 +134,11 @@
 # Disable -fpic with 'make FPIC=' 
 FPIC      = -fpic
 CC        = gcc
-CC_OPTS1  = -O3 -ansi -Isrc/ $(CC_WARNINGS) $(M32) $(FPIC)
+CC_OPTS1  = -O3 -ansi -Isrc/ $(CC_WARNINGS) $(M32) $(FPIC) $(SLKCFLAGS)
 CXX       = g++
 
 # libqhullcpp must be listed before libqhull_r, otherwise libqhull_r pulls in userprintf_r.c
-CXX_OPTS1  = -std=c++98 -O3 -Isrc/ $(CXX_WARNINGS) $(M32) $(FPIC)
+CXX_OPTS1  = -std=c++98 -O3 -Isrc/ $(CXX_WARNINGS) $(M32) $(FPIC) $(SLKCXXFLAGS)
 
 # for shared library link
 CC_OPTS3  =
@@ -203,7 +203,8 @@
 # Default targets for make
 
 all: bin-lib bin/rbox bin/qconvex bin/qdelaunay bin/qhalf bin/qvoronoi bin/qhull bin/testqset \
-     bin/testqset_r qtest bin/user_eg2 bin/user_eg3 bin/user_eg qconvex-prompt
+     bin/testqset_r qtest bin/user_eg2 bin/user_eg3 bin/user_eg qconvex-prompt \
+     lib/libqhull.$(SO)
 
 help:
 	head -n 88 Makefile
@@ -285,6 +286,8 @@
 	cp src/libqhull_r/*.h $(ABS_INCDIR)/libqhull_r
 	cp src/libqhullcpp/*.h $(ABS_INCDIR)/libqhullcpp
 	cp src/qhulltest/*.h $(ABS_INCDIR)/libqhullcpp
+	cd $(ABS_INCDIR) && ln -s libqhull qhull
+	cd $(ABS_INCDIR)/libqhull && ln -s libqhull.h qhull.h
 	for lib in qhullstatic qhullstatic_r qhull_r qhullcpp; \
 	do sed \
 		-e 's#@qhull_VERSION@#$(qhull_VERSION)#' \
@@ -746,6 +749,10 @@
 	-(cd lib/ && ln -f -s libqhull_r.$(SO) libqhull_r.so)
 	-(cd lib/ && ln -f -s libqhull_r.$(SO) libqhull_r.$(SONAME_EXT))
 
+lib/libqhull.$(SO): $(LIBQHULLS_RBOX_OBJS)
+	$(CC) -shared -o $@ $(CC_OPTS3) $^
+	-(cd lib/ && ln -f -s libqhull.$(SO) libqhull.$(SONAME_EXT))
+
 # don't use ../qconvex.	 Does not work on Red Hat Linux
 bin/qconvex: src/qconvex/qconvex.o lib/libqhullstatic.a
 	$(CC) -o $@ $< $(CC_OPTS2) -Llib -lqhullstatic -lm
